Difference between the SUSE version in the API and via Command

## Description

During the issue [DTT2 - Development - Tests reorganization](https://github.com/wazuh/wazuh-qa-automation/issues/4600), it was detected that the version obtained from the API for `OpenSUSE 15` (`arm64` and `amd64`) does not match the version obtained via command line on the instance itself. This is the error:

```console
[2025-12-04 17:53:04,071] [ERROR] [test_basic_info.py] TESTING: There is a mismatch between the OS version and the OS version of the installed agent
[2025-12-04 17:53:04,417] [INFO] [agent.py] TESTING: Uninstalling Agent in agent-opensuse-15-amd64
[2025-12-04 17:53:08,220] [CRITICAL] [testing.py] TESTING: Test basic_info failed with error in agent-linux-suse-15-amd64 hosts.
...
    >           assert agent_version in agent_api_version, logger.error('There is a mismatch between the OS version and the OS version of the installed agent')
    E           AssertionError: None
    E           assert '15.7' in '15-SP7'
```

The value `15-SP7` is obtained by querying the Wazuh API, while the value `15.7` is obtained via a command within the instance itself. This is the list of commands used depending on the Linux system where it is executed:

```python
    __linux_version_cmds = {
        "systemd": {"command": "cat /etc/os-release", "parser": __parse_systemd},
        "fedora": {"command": "cat /etc/fedora-release", "parser": __parse_fedora},
        "centos": {"command": "cat /etc/centos-release", "parser": __parse_centos},
        "amazon": {"command": "cat /etc/system-release", "parser": __parse_amazon},
        "rhel": {"command": "cat /etc/redhat-release", "parser": __parse_rhel},
        "oracle": {"command": "cat /etc/redhat-release", "parser": __parse_rhel},
        "debian": {"command": "cat /etc/debian_version", "parser": lambda x: x},
        "ubuntu": {"command": "cat /etc/lsb-release", "parser": __parse_ubuntu},
    }
```

## Tasks

- Investigate why the API is returning a different value than the value obtained on the machine.
- Determine if it is necessary to modify the value returned by the API.

--- a/src/data_provider/src/osinfo/sysOsParsers.cpp
+++ b/src/data_provider/src/osinfo/sysOsParsers.cpp
@@ -127,16 +127,34 @@ bool UnixOsParser::parseFile(std::istream& in, nlohmann::json& info)
     {
         {"NAME",             "os_name"},
         {"VERSION",          "os_version"},
-        {"VERSION_ID",       "os_version"},
+        {"VERSION_ID",       "version_id"},
         {"ID",               "os_platform"},
         {"BUILD_ID",         "os_build"},
         {"VERSION_CODENAME", "os_codename"}
     };
     const auto ret {parseUnixFile(KEY_MAPPING, SEPARATOR, in, info)};
 
-    if (ret && info.find("os_version") != info.end())
+    if (ret)
     {
-        findMajorMinorVersionInString(info["os_version"], info);
+        if (info.find("os_version") != info.end())
+        {
+            findMajorMinorVersionInString(info["os_version"], info);
+
+            // If VERSION doesn't have major.minor, try VERSION_ID as fallback
+            if ((!info.contains("os_major") || !info.contains("os_minor")) && info.find("version_id") != info.end())
+            {
+                findMajorMinorVersionInString(info["version_id"], info);
+            }
+        }
+        else if (info.find("version_id") != info.end())
+        {
+            // If VERSION is not found, use VERSION_ID as os_version
+            info["os_version"] = info["version_id"];
+            findMajorMinorVersionInString(info["os_version"], info);
+        }
+
+        // Clean up temporary field
+        info.erase("version_id");
     }
 
     return ret;

